spring-integration-core/src/main/java/org/springframework/integration/router/ErrorMessageExceptionTypeRouter.java

```diff
@@ -78,9 +81,12 @@ private void populateClassNameMapping(Set<String> classNames) {
 	@ManagedOperation
 	public void setChannelMapping(String key, String channelName) {
 		super.setChannelMapping(key, channelName);
-		Map<String, Class<?>> newClassNameMappings = new ConcurrentHashMap<>(this.classNameMappings);
-		newClassNameMappings.put(key, resolveClassFromName(key));
-		this.classNameMappings = newClassNameMappings;
+		// TODO: remove application context check in 5.1, here for backwards compatibility
+		if (this.initialized || getApplicationContext() != null) {
```

Comment:
Why do we really need to check here for `AC` ?
The `initialized` flag should be enough: we don't go to load classes until `onInit()` if otherwise.
Why should one worry about class loading if its not a bean yet ?